api/jobserver/badge.go

```diff
@@ -0,0 +1,96 @@
+package jobserver
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/concourse/atc/db"
+)
+
+type badge struct {
+	width     int
+	fillColor string
+	status    string
+}
+
+func (b *badge) statusWidth() int {
+	return b.width - 37
+}
+
+func (b *badge) statusTextWidth() float64 {
+	return float64(b.width)/2 + 17.5
+}
+
+func (b *badge) getSvg() string {
+	const svgTemplate = `<svg xmlns="http://www.w3.org/2000/svg" width="%d" height="20"><linearGradient id="b" x2="0" y2="100%%"><stop offset="0" stop-color="#bbb" stop-opacity=".1"/><stop offset="1" stop-opacity=".1"/></linearGradient><mask id="a"><rect width="%d" height="20" rx="3" fill="#fff"/></mask><g mask="url(#a)"><path fill="#555" d="M0 0h37v20H0z"/><path fill="%s" d="M37 0h%dv20H37z"/><path fill="url(#b)" d="M0 0h%dv20H0z"/></g><g fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="11"><text x="18.5" y="15" fill="#010101" fill-opacity=".3">build</text><text x="18.5" y="14">build</text><text x="%.1f" y="15" fill="#010101" fill-opacity=".3">%s</text><text x="%.1f" y="14">%s</text></g></svg>`
```

Comment:
This should be converted into a proper template using the `template` package. https://golang.org/pkg/text/template/

It would be much nicer if this template was also formatted with newlines. Adding/modifying this template as-is would be very difficult because you'd have to count the number of '%d' and '%s' values. Instead, using the `template` package you can provide a struct that has as its members the fields to be used to fill out the template.